ludwig/models/trainer.py

```diff
@@ -191,8 +236,8 @@ def __init__(
         self.batch_size = batch_size
         self.eval_batch_size = batch_size if eval_batch_size < 1 else eval_batch_size
         self.bucketing_field = bucketing_field
-        self.validation_field = validation_field
-        self.validation_metric = validation_metric
+        self._validation_field = validation_field
```

Comment:
why are _ added here?

Reply:
These needed to be made private so that we could hide the public interface behind a property here:

https://github.com/ludwig-ai/ludwig/blob/backend_refactor/ludwig/models/trainer.py#L819

This is important for the upcoming Ray backend, which needs to override this behavior.